### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by prashantmital**

### CODE REVIEW

File: bindings/python/README.rst [hunk 1]
```diff
@@ -5,4 +5,47 @@ PyMongoArrow
...
        `GitHub <https://github.com/mongodb-labs/mongo-arrow/tree/main/bindings/python>`_
        for the latest source.
+:Documentation: Available at `mongo-arrow.readthedocs.io <https://mongo-arrow.readthedocs.io/en/latest/>`_.
 :Author: Prashant Mital
+
+**PyMongoArrow** is a companion library to PyMongo that contains tools
+for loading MongoDB query result sets as Apache Arrow tables, Pandas
+DataFrames or NumPy arrays.::
+
+  >>> from pymongoarrow.monkey import patch_all
```
Comment (prashantmital) [inline bindings/python/README.rst:15]: This doesn't get syntax highlighted like `requests` README because they use GitHub-flavored markdown

Comment (ShaneHarvey) [reply bindings/python/README.rst:15]: According to https://github.com/github/markup/issues/914#issuecomment-387823670 we can use `.. code-block:: python`.

Comment (prashantmital) [reply bindings/python/README.rst:16]: Doesn't work AFAICT 
image

Comment (ShaneHarvey) [reply bindings/python/README.rst:16]: The syntax is like this:
```
- DataFrames or NumPy arrays.::
+ DataFrames or NumPy arrays.
+
+ .. code-block:: python
+
   >>> from pymongoarrow.monkey import patch_all
```

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**